metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -238,7 +241,7 @@ func NewResourceMetadataEnricher(
 			case *kubernetes.StatefulSet:
 				m[id] = metaGen.Generate(StatefulSetResource, r)
 			case *kubernetes.Namespace:
-				m[id] = metaGen.Generate("namespace", r)
+				m[id] = metaGen.Generate(NamespaceResource, r)
```

Comment:
My only concern is if we break sth here that we dont see. 

My local tests are successful though and for eg.  a pod has the namespace in it

Reply:
I don't think it will break anything. We were following the same logic for all the `case` and we haven't had a problem (at least yet).

But I notice that adding `add_metadata` makes us loose the UUID. I checked for 2 metricsets: `state_deployment` and this new one.

We get an object for every metricset that is like this:

```json
"kubernetes":{
      "deployment":{
         "name":"same-name",
         "uid":"9db79c06-db65-4081-923f-b78deb19eda7"
      },
```

And then we overwrite the object again `kubernetes.deployment` and we loose this data. I am using here `state_deployment` as reference since that one is released, unlike `state_namespace`.